Improve 4Kn support

learmj opened the pull request:
Improves handling of the config variable that defines the target device storage sector size (basically now enforces 512 or 4096)
Encourages mkfs.ext4 to use a minimum block size of 4K if that variable indicates a 4Kn device.

learmj committed:
layer: Restrict sector_size values, clarify description

The capacity validator allowed suffixed values (e.g. "4K"), but
mkfs.vfat -S accepts only plain digits, causing failures.

Restrict sector_size to numeric values (512 or 4096) and update the
description to define it as the target device's logical sector size,
with context on 512n/512e/4Kn.

learmj committed:
image: vfat and ext4 updates for 4Kn device support

mkfs.vfat already uses IGconf_device_sector_size, but try to encourage
mkfs.ext4 to use a minimum block size of 4K if this variable specifies a
4K sector size.
No change to BTFS filesystem creation.

learmj commented:
Hi @martamomotko @timg236 @roliver-rpi 

The only functional difference here is if `IGconf_device_sector_size` is 4096, `-b -4096` is added to the mkfs.ext4 args to ensure a minimum block size of 4K for all ext4 filesystems.

`IGconf_device_sector_size` is already being passed to mkfs.vfat via `-S`.

learmj commented:
Hi @martamomotko @timg236 
From https://github.com/raspberrypi/rpi-eeprom/issues/577, I assume there are no issues if I merge this now?

timg236 commented:
@martamomotko Did you manage to get the rpi-image-gen image to boot on NVMe, it sounded like there was a problem with the MBR  / offset of the FAT

martamomotko commented:
Hi @learmj 
From what I can tell, the 4Kn image generation seems to have incorrect partition alignment. The GPT header offset is at 0x200(512) instead of 0x1000(4096), which means it assumes 512-byte sectors.

learmj committed:
device: Disable 4Kn support

It's not currently possible to create an image compatible with a 4Kn
device using genimage, so restrict device sector size to 512 only.

Refs:
https://github.com/pengutronix/genimage/issues/313

learmj commented:
Hi @martamomotko 
As discussed, we cannot currently use rpi-image-gen to generate 4Kn compatible device images. Therefore setting `IGconf_device_sector_size` to anything other than 512 must result in a configuration validation error.
The other changes on this branch are valid for setting filesystem attributes.
The genimage templates can be revisited once 4Kn support is implemented in upstream genimage.

martamomotko commented:
Yes, thank you @learmj
